code/go/0chain.net/smartcontract/zcnsc/stakepool.go

```diff
@@ -0,0 +1,313 @@
+package zcnsc
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/state"
+	"0chain.net/chaincore/transaction"
+	"0chain.net/core/common"
+	"0chain.net/core/datastore"
+	"0chain.net/core/encryption"
+	"0chain.net/core/util"
+	"0chain.net/smartcontract/stakepool"
+	"0chain.net/smartcontract/stakepool/spenum"
+)
+
+// unlock response
+type unlockResponse struct {
+	// one of the fields is set in a response, the Unstake if can't unstake
+	// for now and the TokenPoolTransferResponse if it has a pool had unlocked
+	Unstake bool          `json:"unstake"` // max time to wait to unstake
+	Balance state.Balance `json:"balance"`
+}
+
+type stakePoolRequest struct {
+	PoolID       string `json:"pool_id,omitempty"`
+	AuthorizerID string `json:"authorizer_id,omitempty"`
+}
+
+func (spr *stakePoolRequest) decode(p []byte) (err error) {
+	if err = json.Unmarshal(p, spr); err != nil {
+		return
+	}
+	return // ok
+}
+
+// ----------- LockingPool pool --------------------------
+
+type StakePool struct {
```

Comment:
Why do we need to wrap the stakePool.StakePool object?

For example, if we need new methods can't we
```go
var stakePool stakepool.StakePool
```
work as there are no new fields. Unlike [storagesc.stakePool ](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/stakepool.go#L49) where there are two extra fields.

Reply:
Ok

Reply:
I'd prefer leaving this because we have to implement utils.MPTSerializable with many other involved just repeating the code. Composite inheritance helps here to get already implemented functions for free